packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -22,45 +32,139 @@ const sidebarStyles = css({
   height: '100%',
 });
 
-// TODO We will get rid of this placeholder when expose the necessary props outside
-// eslint-disable-next-line
-const noop_tmp = (() => {}) as any;
+type ConnectionErrorToastBodyProps = {
+  info: ConnectionInfo | null;
+  onReview: () => void;
+};
+
+const connectionErrorToastBodyStyles = css({
+  display: 'flex',
+  alignItems: 'start',
+  gap: spacing[2],
+});
 
+const connectionErrorToastActionMessageStyles = css({
+  marginTop: spacing[1],
+  flexGrow: 0,
+});
+
+function ConnectionErrorToastBody({
+  info,
+  onReview,
+}: ConnectionErrorToastBodyProps): React.ReactElement {
+  return (
+    <span className={connectionErrorToastBodyStyles}>
+      <span>
+        There was a problem connecting{' '}
+        {info ? `to ${getConnectionTitle(info)}` : ''}
+      </span>
+      {info && (
+        <Link
+          className={connectionErrorToastActionMessageStyles}
+          hideExternalIcon={true}
+          onClick={onReview}
+        >
+          REVIEW
+        </Link>
+      )}
+    </span>
+  );
+}
 // Having props here is useful as a placeholder and we will fix it with the first props.
 // eslint-disable-next-line
-export function MultipleConnectionSidebar({}: MultipleConnectionSidebarProps) {
+export function MultipleConnectionSidebar({
+  appName,
+}: MultipleConnectionSidebarProps) {
+  const { openToast, closeToast } = useToast('multiple-connection-status');
+  const cancelCurrentConnectionRef = useRef<(id: string) => Promise<void>>();
+
+  const [isExpanded, setIsExpanded] = useState(true);
+  const [isConnectionFormOpen, setIsConnectionFormOpen] = useState(false);
+
+  const onConnected = useCallback(() => {
+    closeToast('connection-status');
```

Comment:
This probably should have an id based on a connection info id, otherwise one connection closes toasts for other connections. As the process is async and is not even visually blocked by the modal of any kind, this is very confusing: you can see progress or errors for multiple ongoing connections